viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
+            return 0.0
+
+    for movie in watched_list:
+        ratings_sum += movie["rating"]
+        average = ratings_sum / len(watched_list)
+    return average
 
+def get_most_watched_genre(user_data):
+    watched_list = user_data["watched"]
+    genre_list = []
 
+    if watched_list == []:
+        return None
+    
+    for movie in watched_list:
+        genre_list.append(movie["genre"])
+    
+    top_genre = max(set(genre_list), key=genre_list.count)
+    return top_genre
+
+        
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_movies = []
+    friend_movies_list = []
 
-        
+    #creating a dictionary of friend's watched movies and putting into a list
+    for watchlist in user_data["friends"]:
+        for watched in watchlist.values():
+            for movies_dict in watched:
+                friend_movies_list.append(movies_dict)
+    
+    #comparing user list and friends list for unique titles
+    for movies in user_data["watched"]:
+        if movies not in friend_movies_list:
+            unique_movies.append(movies)
+    #returning list of movies seen by user but not by friends
+    return unique_movies
+
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+    user_movies_list = []
+    return_list = []
+    
+    #creating a dictionary of user's watched and putting into a list
+    for movies in user_data["watched"]:
+        user_movies_list.append(movies)
+
+    #comparing friends list and user list for unique titles
+    for watchlist in user_data["friends"]:
+        for watched in watchlist.values():
+            for movies_dict in watched:
+                if movies_dict not in user_movies_list:
+                    unique_movies.append(movies_dict)
+    
+    #checking unique list for any duplicates 
+    for movie in unique_movies:
+        if movie not in return_list:
+            return_list.append(movie)
+    #returning list of movies that have been seen by friends but not user
+    return return_list         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    user_unseen_movies = get_friends_unique_watched(user_data)
+    recommendations = []  
+
+    if user_data['watched'] == []:
+        return recommendations
+    
+    for movies in user_unseen_movies:
+        for subscription in user_data["subscriptions"]:
+            if subscription == movies['host']:
+                recommendations.append(movies)
+    
+    return recommendations
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    rec = get_available_recs(user_data)
+    top_genre = get_most_watched_genre(user_data)

```

Comment:
Using `get_available_recs` here will pass all of the tests, but the wave directions don't specify that the subscription should be factored into the calculation.  This is a really clean implementation, but this function should return recommendations regardless of the subscription status.